internal/heartbeat/heartbeat.go

```diff
@@ -195,12 +230,19 @@ func (s *Heartbeat) Init(config models.DeviceConfigurationMessage) error {
 }
 
 func (s *Heartbeat) Update(config models.DeviceConfigurationMessage) error {
+	s.tickerLock.Lock()
```

Comment:
why not using defer to unlock it?

Reply:
I have to lock it to avoid having race with `s.previousPeriodSeconds`
This will be a ReadLock when I'll be done implementing the change using RWMutex